src/main/java/io/airlift/slice/Slice.java

```diff
@@ -696,6 +732,105 @@ public int indexOfByte(int b)
     }
 
     /**
+     * Returns the index of the first occurrence of the pattern with this slice.
+     * If the pattern is not found -1 is returned If patten is empty, the offset
+     * is returned.
+     */
+    public int indexOf(Slice slice)
+    {
+        return indexOf(slice, 0);
+    }
+
+    /**
+     * Returns the index of the first occurrence of the pattern with this slice.
+     * If the pattern is not found -1 is returned If patten is empty, the offset
+     * is returned.
+     */
+    public int indexOf(Slice pattern, int offset)
+    {
+        if (pattern.length() == 0) {
+            return min(offset, size);
+        }
+
+        if (size == 0 || offset >= size) {
+            return -1;
+        }
+
+        // Do we have enough characters
+        if (pattern.length() < SIZE_OF_INT || size < SIZE_OF_LONG) {
```

Comment:
I understand using fast match for pattern whose `length` is less than `SIZE_OF_INT` can cause segfault. But it feels weird to me that the whole optimization is dropped as a result.